quora-api/src/main/java/com/upgrad/quora/api/controller/AdminController.java

```diff
@@ -0,0 +1,34 @@
+package com.upgrad.quora.api.controller;
+
+import com.upgrad.quora.api.model.UserDeleteResponse;
+import com.upgrad.quora.service.business.AuthenticationService;
+import com.upgrad.quora.service.exception.AuthorizationFailedException;
+import com.upgrad.quora.service.exception.UserNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/")
+public class AdminController {
+
+    @Autowired
+    AuthenticationService authenticationService;
+
+    /** This end-point is used to remove a user from Quora db
+     *
+     * @param authorization
+     * @param userId UUID of the user who has to be deleted from db
+     * @return UUID of the deleted user
+     * @throws AuthorizationFailedException unAuthorization exception is thrown when non admin user tries to delete the User
+     * @throws UserNotFoundException thrown when trying to delete un-registered user
+     */
+    @RequestMapping(method = RequestMethod.DELETE, path = "/admin/user/{userId}", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public ResponseEntity<UserDeleteResponse> deleteUser(@RequestHeader("authorization") final String authorization, @PathVariable String userId) throws AuthorizationFailedException, UserNotFoundException {
+        String deletedUserUUID = authenticationService.checkUserByAuthToken(authorization, userId);
```

Comment:
Can the service name be changed as deleteAuthUser() to make it more related to the function it does

Reply:
There are 2 table's that get affect by this
1. UserEntity (users)
2. User Auth table
so made it generic as delete user as it does not depend only on User auth table.

Do you want me to go head and update it please let me know